### PR TITLE
Require to opt-in for the global service account

### PR DESCRIPTION
**Pull request by RichardChen820**
Require to opt in the global service account

### CODE REVIEW

File: deploy/parameter/helm-values.yaml [hunk 1]
```diff
@@ -20,7 +20,7 @@ requestTracing:
 
 workloadIdentity:
   enabled: true
-  disableGlobalServiceAccount: false
+  enableGlobalServiceAccount: false
```
Comment (zhenlan) [inline deploy/parameter/helm-values.yaml:23]: This is a verb. As a property name, it should be `globalServiceAccountEnabled`. Alternatively, you can have `useGlobalServiceAccount` or `globalServiceAccountAllowed`. Any preference?

Comment (RichardChen820) [reply deploy/parameter/helm-values.yaml:23]: globalServiceAccountEnable is good to me

File: internal/controller/utils.go [hunk 2]
```diff
@@ -247,14 +247,14 @@ func verifyWorkloadIdentityParameters(workloadIdentity *acpv1.WorkloadIdentityPa
 	var authCount int = 0
 
 	if workloadIdentity.ManagedIdentityClientId != nil {
-		if strings.EqualFold(os.Getenv(WorkloadIdentityDisableGlobalServiceAccount), "true") {
+		if strings.EqualFold(os.Getenv(WorkloadIdentityGlobalServiceAccountEnabled), "false") {
```
Comment (zhenlan) [inline internal/controller/utils.go:250]: Why are we reading it from the environment? I thought it's a parameter passed in.

Comment (RichardChen820) [reply internal/controller/utils.go:250]: The gap is how the controller can know the global service account is enabled or disabled, it just know it via the environment variable that been set while installation.

**Review by zhenlan — APPROVED** [review]

**Pull request merged**